nightly: FUTURE_SEQUANA does not build - reset reason error

<!--

   ************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure all sections of the template below are filled in and no changes 
   are made to the template format. Only bugs should be raised here as issues. 
   Questions or enhancements should instead be raised on our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

-->

### Description of defect

<!--
    Add detailed description of what you are reporting.
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html
-->

Error:

```
[Error] @0,0: L6218E: Undefined symbol hal_reset_reason_clear (referred from BUILD/FUTURE_SEQUANA/ARM-DEVELOP/mbed-os/drivers/source/ResetReason.o).
[Error] @0,0: L6218E: Undefined symbol hal_reset_reason_get (referred from BUILD/FUTURE_SEQUANA/ARM-DEVELOP/mbed-os/drivers/source/ResetReason.o).
[Error] @0,0: L6218E: Undefined symbol hal_reset_reason_get_raw (referred from BUILD/FUTURE_SEQUANA/ARM-DEVELOP/mbed-os/drivers/source/ResetReason.o).
Warning: L3912W: Option 'legacyalign' is deprecated.
Error: L6218E: Undefined symbol hal_reset_reason_clear (referred from BUILD/FUTURE_SEQUANA/ARM-DEVELOP/mbed-os/drivers/source/ResetReason.o).
Error: L6218E: Undefined symbol hal_reset_reason_get (referred from BUILD/FUTURE_SEQUANA/ARM-DEVELOP/mbed-os/drivers/source/ResetReason.o).
Error: L6218E: Undefined symbol hal_reset_reason_get_raw (referred from BUILD/FUTURE_SEQUANA/ARM-DEVELOP/mbed-os/drivers/source/ResetReason.o).
Finished: 0 information, 1 warning and 3 error messages.
[ERROR] Warning: L3912W: Option 'legacyalign' is deprecated.
Error: L6218E: Undefined symbol hal_reset_reason_clear (referred from BUILD/FUTURE_SEQUANA/ARM-DEVELOP/mbed-os/drivers/source/ResetReason.o).
Error: L6218E: Undefined symbol hal_reset_reason_get (referred from BUILD/FUTURE_SEQUANA/ARM-DEVELOP/mbed-os/drivers/source/ResetReason.o).
Error: L6218E: Undefined symbol hal_reset_reason_get_raw (referred from BUILD/FUTURE_SEQUANA/ARM-DEVELOP/mbed-os/drivers/source/ResetReason.o).
Finished: 0 information, 1 warning and 3 error messages.
```

I believe this is the offending PR https://github.com/ARMmbed/mbed-os/pull/12440. 

cc @ARMmbed/team-cypress @maclobdell Please review

#### Target(s) affected by this defect ?

FUTURE_SEQUANA + M0 as well

#### Toolchain(s) (name and version) displaying this defect ?

All

#### What version of Mbed-os are you using (tag or sha) ?
<!--
    For a released version please provide the release tag (this can be found as per the instructions below)

    mbed-os version can be found in /platform/mbed_version.h. The tag can be reconstructed as follows:
    mbed-os-MBED_MAJOR_VERSION.MBED_MINOR_VERSION.MBED_PATCH_VERSION
 
    Master branch is indicated by 'mbed-os-99.99.99
    
    For an issue found on Master please provide the sha being used.
-->

mbed-os-99.99.99

#### What version(s) of tools are you using. List all that apply (E.g. mbed-cli)

Not applicable

#### How is this defect reproduced ? 

Build blinky for this target

--- a/targets/targets.json
+++ b/targets/targets.json
@@ -14014,7 +14014,8 @@
             "I2CSLAVE",
             "USBDEVICE",
             "QSPI",
-            "WATCHDOG"
+            "WATCHDOG",
+            "RESET_REASON"
         ],
         "macros_add": [
             "CY8C6347BZI_BLD53"
@@ -14074,7 +14075,8 @@
             "I2CSLAVE",
             "USBDEVICE",
             "QSPI",
-            "WATCHDOG"
+            "WATCHDOG",
+            "RESET_REASON"
         ],
         "macros_add": [
             "CY8C6347BZI_BLD53"

